RecoMuon/MuonRechitClusterProducer/plugins/RechitClusterProducer.cc

```diff
@@ -0,0 +1,268 @@
+    }
+    recIt++;
+  }
+  fastjet::ClusterSequence clus_seq(fjInput, jet_def);
+
+  //keep all the clusters
+  double ptmin = 0.0;
+  std::vector<fastjet::PseudoJet> fjJets = fastjet::sorted_by_pt(clus_seq.inclusive_jets(ptmin));
+
+  auto clusters = std::make_unique<RecHitClusterCollection>();
+  if (!fjJets.empty()) {
+    for (unsigned int ijet = 0; ijet < fjJets.size(); ++ijet) {
+      // get the fastjet jet
+      const fastjet::PseudoJet& fjJet = fjJets[ijet];
+
+      // skip if the cluster has too few rechits
+      if (int(fjJet.constituents().size()) < nRechitMin_)
+        continue;
+      // get the constituents from fastjet
+      RecHitRefVector rechits;
+      for (unsigned int i = 0; i < fjJet.constituents().size(); i++) {
+        auto index = fjJet.constituents()[i].user_index();
+        if (index >= 0 && static_cast<unsigned int>(index) < inputs_.size()) {
+          rechits.push_back(inputs_[index]);
+        }
+      }
+
+      //Derive cluster properties
+      int nStation = 0;
+      int totStation = 0;
+      float avgStation = 0.0;
+      std::map<int, int> station_count_map;
+      for (auto& rechit : rechits) {
+        station_count_map[RecHitTrait.station(*rechit)]++;
+      }
+      //station statistics
+      std::map<int, int>::iterator it;
+      for (auto const& [station, count] : station_count_map) {
+        if (count >= nStationThres_) {
+          nStation++;
+          avgStation += (station) * (count);
+          totStation += (count);
+        }
+      }
+      if (totStation != 0) {
+        avgStation = avgStation / totStation;
+      }
+      float invN = 1.f / rechits.size();
+      // cluster position is the average position of the constituent rechits
+      float jetX = fjJet.px() * invN;
+      float jetY = fjJet.py() * invN;
+      float jetZ = fjJet.pz() * invN;
+
+      math::RhoEtaPhiVectorF position(
+          std::sqrt(jetX * jetX + jetY * jetY), etaFromXYZ(jetX, jetY, jetZ), std::atan2(jetY, jetX));
+      Trait::emplace_back(clusters.get(), position, nStation, avgStation, rechits);
+    }
+  }
+  ev.put(std::move(clusters));
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+template <typename Trait>
+void RechitClusterProducerT<Trait>::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<int>("nRechitMin", 50);
+  desc.add<double>("rParam", 0.4);
+  desc.add<int>("nStationThres", 10);
+  desc.add<edm::InputTag>("recHitLabel", edm::InputTag(Trait::recHitLabel()));
+  descriptions.add(Trait::producerName(), desc);
+}
+struct DTRecHitTrait {
+  using GeometryType = DTGeometry;
+  using InputType = DTRecHitCollection;
+  using RecHitRef = edm::Ref<DTRecHitCollection>;
+  using RecHitRefVector = edm::RefVector<DTRecHitCollection>;
+  static std::string recHitLabel() { return "dt1DRecHits"; }
+  static std::string producerName() { return "DTrechitClusterProducer"; }
```

Comment:
```suggestion
  static std::string producerName() { return "dtRechitClusterProducer"; }
```
instances are not capitalized